core/src/main/resources/lib/hudson/buildLink.jelly

```diff
@@ -52,7 +52,7 @@ THE SOFTWARE.
         </j:when>
         <j:otherwise>
           <a href="${attrs.href ?: rootURL+'/'+r.url}" class="model-link">
-          <l:icon alt="${r.iconColor.description}" class="${r.buildStatusIconClassName} icon-sm" style="margin-left: 0; position: relative; top: -0.1rem;"/><span class="jenkins-icon-adjacent">${jobName_}#<!-- -->${number}</span></a>
+          <l:icon alt="${r.iconColor.description}" src="symbol-status-${r.iconColor.iconName}" class="icon-sm" style="margin-left: 0; position: relative; top: -0.1rem;"/><span class="jenkins-icon-adjacent">${jobName_}#<!-- -->${number}</span></a>
```

Comment:
This breaks the build link again (3 time in the last past months) that is shown in the warnings and coverage links. Maybe we should provide a UI tests for this feature.

<img width="432" alt="Bildschirmfoto 2023-12-23 um 17 16 32" src="https://github.com/jenkinsci/jenkins/assets/503338/ba736562-4b88-47f9-8083-c3df180e52d1">